content/en/docs/tasks/job/indexed-parallel-processing-static.md

```diff
@@ -0,0 +1,166 @@
+---
+title: Indexed Job for Parallel Processing with Static Work Assignment
+content_type: task
+min-kubernetes-server-version: v1.21
+weight: 30
+---
+
+{{< feature-state for_k8s_version="v1.21" state="alpha" >}}
+
+<!-- overview -->
+
+{{< note >}}
+To be able to create Indexed Jobs, make sure to enable the `IndexedJob`
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
+on the [API server](docs/reference/command-line-tools-reference/kube-apiserver/)
+and the [controller manager](/docs/reference/command-line-tools-reference/kube-controller-manager/).
+{{< /note >}}
+
+In this example, we will run a Kubernetes Job with multiple parallel worker processes.
+Each worker Pod gets assigned a unit of work statically, which is determined by
+the pod index available in the {{< glossary_tooltip text="annotation" term_id="annotation" >}}
+`batch.alpha.kubernetes.io/job-completion-index`, also accessible to containers
+in the `JOB_COMPLETION_INDEX` environment variable.
+
+Here is an overview of the steps in this example:
+
+1. **Create an image that can read the pod index**
+2. **Start an Indexed Job**. The downward API allows you to pass the annotation
+   as an environment variable or file to the container.
+
+<!-- steps -->
+
+## Create an Image
+
+To access the work item from the worker program, you have a few options:
+
+1. Read the `JOB_COMPLETION_INDEX` environment variable. The Job controller
+   automatically links this variable to the annotation containing the completion
+   index.
+1. Read a file that contains the completion index.
+1. Assuming that you can't modify the program, you can wrap it with a script
+   that reads the index using any of the methods above and converts it into
+   something that the program can use as input.
+  
+In this example, we show option 3. The following program knows how to
+accept a work item from a command line argument.
+
+{{< codenew language="python" file="application/job/worker.py" >}}
+
+Give the script execution permission:
+
+```shell
+chmod +x worker.py
+```
+
+Now, build an image. Download the [Dockerfile](/examples/application/job/Dockerfile)
+and build the image with this command:
+
+```shell
+docker build -t job-worker .
+```
+
+For the [Docker Hub](https://hub.docker.com/), tag your app image with
+your username and push to the Hub with the below commands. Replace
+`<username>` with your Hub username.
+
+```shell
+docker tag job-worker <username>/job-worker
+docker push <username>/job-worker
+```
+
+If you are using [Google Container Registry](https://cloud.google.com/tools/container-registry/),
+tag your app image with your project ID, and push to GCR. Replace `<project>`
+with your project ID.
+
+```shell
+docker tag job-wq-1 gcr.io/<project>/job-worker
+gcloud docker -- push gcr.io/<project>/job-worker
+```
```

Comment:
This might not be in quite line with our content guide:
https://kubernetes.io/docs/contribute/style/content-guide/#third-party-content

Consider using [tabs](https://kubernetes.io/docs/contribute/style/hugo-shortcodes/#tabs) to document image push. That way if someone wants to add a tab for another kind of image registry then, within reason, they easily can. Order the tabs alphabetically.